app/lib/text/renderer.rb

```diff
@@ -17,14 +17,27 @@ def render_string(template)
       else
         rendered = template
         rendered = render_report(rendered)
-        @text_component.sensors.each do |sensor|
-          s = SensorDecorator.new(sensor)
-          rendered.gsub!(/({\s*#{ Regexp.quote("value(#{s.id})") }\s*})/, s.last_value(@opts))
+        
+        rendered.scan(/{\s*value\(\s*(\d+)\s*\)\s*}/).each do |sensor_id|
```

Comment:
I have undone most of this change as it hasn’t been well thought out. There are a couple of test for the text render for all possible scenarios. Are you missing anything specific?